scenarios.md

```diff
@@ -0,0 +1,199 @@
+
+- The local developer has access to signing keys
+- The local environment has a policy by which it states the set of keys it accepts.
+- The signing and validation of artifacts does not require a registry.
+- The signature used for validation may be hosted in a registry, or other accessible location.
+- The lack of a registry name does not infer docker.io as the default registry. This does imply changes to either the docker client, the notary client or both.
+
+### Scenario #2: Sign, Rename, Push, Validate
+
+Once the developer has locally validated the build, sign, validate scenario, they will likely want to push the artifact to a registry used for deployment.
+
+1. Locally build and sign an artifact, such as a container image
+1. Rename the artifact to reflect the registry it will be pushed to:  
+  `$ docker tag hello-world:1 myregistry.domain.com/marketing/web:1`
+  `$docker push myregistry.domain.com/marketing/web:1`
+1. Deploy the artifact to a cluster that requires signatures:  
+  `$kubectl apply -f kubedeploy.yaml`
+  > **Note:** the above example presumes `kubedeploy.yaml` references the myregistry.domain.com/marketing/web:1 image
+
+**Implications of this requirement:**
+
+- Signatures can be validated based on the tag referenced, however the signature is not tied to a specific name. The artifact can be renamed without invalidating the signature.
+- This does not preclude tag locking scenarios, where a registry can provide users with the ability to lock a tag to a specific. If a tag is updated with another signed artifact, the tag is considered signed and follows the signature validation rules of the new signature. For example, if a base image `fx:1.0` is signed, and rebuilt with a patched version, the updated `fx:1.0` is new content, and a valid scenario.
+- If the tag for `fx:1.0` is redirected to another unsigned manifest, the signature validation of `fx:1.0` will fail as it's no longer signed. This is no different than signed binaries being updated or replaced on a users local computer.
+
+### Scenario #3: Automate Build, Sign, Push
+
+A CI system is triggered by a git commit. The system can build the artifacts, sign them, and push to a registry. The production system can pull the artifacts, verify the signatures and run them.
+
+1. A CI solution responds to a git commit notification
+1. The CI system clones the git repo and builds the artifacts, with fully qualified names: **image**: `myregistry.domain.com/sample/hello-world:aa31` and a **helm chart**: `myregistry.domain.com/sample/charts/hello-world:aa31`
+1. The CI system signs the artifact with locally available keys *
+  (**Note:** key management deferred from this requirement)*
+1. The artifact is pushed to a registry:  
+  `$ docker push myregistry.domain.com/sample/hello-world:aa31`
+  `$ helm chart push myregistry.domain.com/sample/charts/hello-world:aa31`
+1. The artifacts are deployed, referencing the registry they were pushed to.
```

Comment:
I'm not sure what this means -- ideally the artifacts (at least the images) wouldn't reference any registry, since that would break content-addressability. Are you saying the helm chart references the image? Or the deployment system references the registry?

Reply:
I re-read with your question in mind and realized "referencing the registry they were pushed to" was vague. While helm, as a deployment artifact, does need to reference the artifacts being deployed, that details wasn't the intent.
Based on today's call, I've started changing references to "the orchestrator".